### ISSUE TITLE
docker-compose: EACCES: permission denied, mkdir '/opt/node_app/app/node_modules/.cache'

### ISSUE THREAD

**Issue by jdreo** [oncodash/oncodash#69]
# Summary

When following the install process, after starting `docker-compose up`, the nodeserver complains about not being able to write in a cache dir in a mounted point.

# Bug full description

Build with:
```sh
$ docker-compose build && docker-compose run --rm backend sh -c "python manage.py makemigrations" && docker-compose run --rm backend sh -c "python manage.py migrate" && docker-compose run --rm backend sh -c "python manage.py flush --no-input" && docker-compose run --rm backend sh -c "python manage.py fake_data -n 10"
$ docker-compose up
```
Output excerpt:
```
oncodash-nodeserver-1  | Starting the development server...
oncodash-nodeserver-1  | 
oncodash-nodeserver-1  | Failed to compile.
oncodash-nodeserver-1  | 
oncodash-nodeserver-1  | EACCES: permission denied, mkdir '/opt/node_app/app/node_modules/.cache'
oncodash-nodeserver-1  | /opt/node_app/node_modules/watchpack/lib/chokidar.js:17
oncodash-nodeserver-1  | throw new Error(
oncodash-nodeserver-1  | ^
oncodash-nodeserver-1  | 
oncodash-nodeserver-1  | Error: No version of chokidar is available. Tried chokidar@2 and chokidar@3.
oncodash-nodeserver-1  | You could try to manually install any chokidar version.
oncodash-nodeserver-1  | chokidar@3: Error: Cannot find module 'chokidar'
```

**Comment by jdreo** [comment]
The problem seems to come from the bypassed mount point of the node_module directory in the container definition file `docker-compose.yml`:
```yaml
        # this is a workaround to prevent host node_modules from accidently
        # getting mounted in container in case you want to use node/npm both
        # outside container for test/lint etc. and also inside container this
        # will overwrite the default node_modules dir in container so it won't
        # conflict with our /opt/app/node_modules location.
       - type: volume
         source: notused
         target: /opt/node_app/app/node_modules
```
If commented out, there is no more the mentioned error.

**Comment by jdreo** [comment]
Even after rebuilding form scratch? On my setup this leads to another error
(but at the same location).

Le lun. 17 oct. 2022 à 16:12, vittoriopipoli ***@***.***> a
écrit :

**Comment by prittt** [comment]
Solved with #71.
